RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc

```diff
@@ -140,12 +139,11 @@ void SeedGeneratorFromProtoTracksEDProducer::produce(edm::Event& ev, const edm::
         GlobalTrackingRegion region(mom_perp, vtx, 0.2, 0.2);
 
         seedCreator_.init(region, es, nullptr);
-        seedCreator_.makeSeed(
-            *result,
-            SeedingHitSet(hits[0],
-                          hits[1],
-                          hits.size() > 2 ? hits[2] : SeedingHitSet::nullPtr(),
-                          (includeFourthHit_ && hits.size() > 3) ? hits[3] : SeedingHitSet::nullPtr()));
+
+        if (hits.size() == 4 and not includeFourthHit_)
```

Comment:
should we not just obsolete ```not includeFourthHit_```?

Reply:
I'd say yes, but I think this is still used somewhere and would need a cleanup that I'd leave to another PR.

Reply:
based on https://github.com/cms-sw/cmssw/pull/36184#issuecomment-1009989757
I guess that this line should change to 
```suggestion
        if (hits.size() > 3 and not includeFourthHit_)
```
to be consistent with the old implementation.
This is mainly to keep this PR technical.